### PR TITLE
Add solution outline

### PR DESCRIPTION
**Pull request by amulware**
Not much to say, it is what it is.

Squash at your own pleasure.

### CODE REVIEW

File: docs/solution-outline.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+It contains several simple helper types, as well as shared model interfaces.
+
+
+## Ultrawave
+
+Ultrawave provides a thin layer around the public Toggl Api.
+
+Dependencies: Multivac
+
+It's main responsibility is calling the correct Http endpoints, and (de)serializing models from/for the api.
```
Comment (willsb) [inline docs/solution-outline.md]: Its*

File: docs/solution-outline.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+To allow for the latter, it has limited knowledge of Toggl's business logic in the sense that it will only serialise model properties valid for a specific endpoint and given business logic specific constraints.
+
+
+## PrimeRadiant
+
+PrimeRadiant is our local storage (database) project.
+
+Dependencies: Multivac
+
+This project contains interfaces representing the local storage repository of a Toggl app. These interfaces can be implemented in additional projects corresponding to specifics database/storage frameworks/libraries.
```
Comment (willsb) [inline docs/solution-outline.md]: Also contains models, maybe worth mentioning

File: docs/solution-outline.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+Next to containing business logic, this project also ties together the Toggl Api and local storage with a comprehensive syncing algorithm.
+This is the _foundation_ for any of our Toggl mobile apps, though it contains no mobile specific code itself, and can be used to run a Toggl app on any platform.
+
+
+## Foundation.MvvmCross
+
+Foundation.MvvmCross is a shared MvvmCross view model library.
+
+Dependencies: Multivac, Foundation
```
Comment (willsb) [inline docs/solution-outline.md]: Isn't the dependency on Multivac infered from the dependency on Foundation?

Comment (amulware) [inline docs/solution-outline.md]: I figured that maybe not all of the projects depend on it explicitely, so I wanted the option to leave it out (see Daneel right now). But if you think this doesn't add much, I'm alright simplifying it. Let me know.

File: docs/solution-outline.md (hunk 2 above)
Comment (amulware) [inline docs/solution-outline.md]: Agreed.

File: docs/solution-outline.md (hunk 1 above)
Comment (amulware) [inline docs/solution-outline.md]: Duh.

**Comment by amulware** [comment]
Ludic or lucid?

**Comment by willsb** [comment]
I meant ludic, BTW :smile:

**Pull request merged**